File: 2022/Python/day_1.py
import pandas as pd
import numpy as np
from numpy import NaN as nan
from itertools import groupby

data_d1 = pd.read_csv('data/day_1.csv', header = None)
calories = np.array(data_d1[0])

by_elf = [np.sum(list(v)) for k, v in groupby(calories, np.isfinite) if k]

max_calories = max(by_elf)

# part 2: top 3 elves
top_3_sum = np.sum(sorted(by_elf, reverse = True)[0:3])

File: 2022/R/day_1.R
# Day 1
# How many Calories are being carried by the Elf carrying the most Calories?
library(dplyr)

data_d1 <- read.csv('2022/data/day_1.csv', header = FALSE) 

data_by_elf <- data_d1 %>% 
  rename(calories = V1) %>% 
  mutate(elf = row_number()) %>% 
  mutate(elf = ifelse((is.na(calories)|elf == 1), as.character(elf), NA)) %>% 
  fill(elf) %>% 
  filter(!is.na(calories)) 

totals <- data_by_elf %>% 
  group_by(elf) %>%
  summarise(total_calories = sum(calories))
  
most = max(totals$total_calories)

